src/m365/planner/commands/task/task-checklistitem-remove.ts

```diff
@@ -0,0 +1,124 @@
+}
+
+class PlannerTaskChecklistItemRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_CHECKLISTITEM_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the checklist item from the planner task';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'isChecked'];
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.confirm = (!(!args.options.confirm)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.confirm) {
+      this.removeChecklistitem(logger, args, cb);
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the checklist item with id ${args.options.id} from the planner task?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          this.removeChecklistitem(logger, args, cb);
+        }
+      });
+    }
+  }
+
+  private removeChecklistitem(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    this
+      .getTaskDetailsEtag(args.options.taskId)
+      .then(etag => {
+        const requestOptionsTaskDetails: any = {
+          url: `${this.resource}/v1.0/planner/tasks/${args.options.taskId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            checklist: {
+              [args.options.id]: null
+            }
+          }
+        };
+
+        return request.patch(requestOptionsTaskDetails);
+      })
+      .then((res: any): void => {
+        if (!args.options.output || args.options.output === 'json') {
+          logger.log(res.checklist);
+        }
+        else {
+          //converted to text friendly output
+          const output = Object.getOwnPropertyNames(res.checklist).map(prop => ({ id: prop, ...(res.checklist as any)[prop] }));
+          logger.log(output);
+        }
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getTaskDetailsEtag(taskId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/tasks/${encodeURIComponent(taskId)}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        const etag: string | undefined = response ? response['@odata.etag'] : undefined;
+        if (!etag) {
+          return Promise.reject(`Error fetching task details`);
```

Comment:
If the call fails, I think it will reject the promise. 

So if you want to return a specific message when the task cannot be found, you should use the `.catch` callback of the promise.

Reply:
Thank you. I have implemented the changes. Can you please review?